conn_http.go

```diff
@@ -93,32 +145,103 @@ func (h *httpConnect) executeRequest(ctx context.Context, req *http.Request) (io
 		return nil, err
 	}
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("clickhouse [execute]:: got no 200 code('%d')", resp.StatusCode)
+
+		msg, err := readResponse(resp)
+
+		if err != nil {
+			return nil, fmt.Errorf("clickhouse [execute]:: failed to read the response (status code %d): %w", resp.StatusCode, err)
+		}
+
+		return nil, fmt.Errorf("clickhouse [execute]:: %d code: %s", resp.StatusCode, string(msg))
 	}
 
 	return resp.Body, nil
 }
 
-func (h *httpConnect) ping(ctx context.Context) error {
-	req, err := h.prepareRequest(ctx, "SELECT 1")
+func (h *httpConnect) exec(ctx context.Context, query string, args ...interface{}) error {
+	query, err := bind(h.location, query, args...)
 	if err != nil {
 		return err
 	}
 
-	res, err := h.executeRequest(ctx, req)
+	req, err := h.prepareRequest(ctx, strings.NewReader(query), nil)
 	if err != nil {
 		return err
 	}
-	s, err := io.ReadAll(res)
+
+	res, err := h.executeRequest(ctx, req)
+	if res != nil {
+		defer res.Close()
+		// we don't care about result, so just discard it to reuse connection
+		_, _ = io.Copy(ioutil.Discard, res)
+	}
+
+	return err
+}
+
+func (h *httpConnect) query(ctx context.Context, query string, args ...interface{}) (*rows, error) {
```

Comment:
can we move to conn_http_query.go to be consistent with native?

Reply:
Yes, moved